src/gamecontroller.cs

```diff
@@ -535,4 +621,1162 @@ interface GCEventViewController {
+	}
+
+	[TV (14, 0), Mac (11, 0), iOS (14, 0)]
+	[BaseType (typeof (NSObject))]
+	interface GCKeyboard : GCDevice
+	{
+		[NullAllowed, Export ("keyboardInput", ArgumentSemantic.Strong)]
+		GCKeyboardInput KeyboardInput { get; }
+
+		[Static]
+		[NullAllowed, Export ("coalescedKeyboard", ArgumentSemantic.Strong)]
+		GCKeyboard CoalescedKeyboard { get; }
+
+		[Notification, Field ("GCKeyboardDidConnectNotification")]
+		NSString DidConnectNotification { get; }
+
+		[Notification, Field ("GCKeyboardDidDisconnectNotification")]
+		NSString DisconnectNotification { get; }
+	}
+
+	delegate void GCKeyboardValueChangedHandler (GCKeyboardInput keyboard, GCControllerButtonInput key, nint keyCode, bool pressed);
+
+	[TV (14,0), Mac (11,0), iOS (14,0)]
+	[BaseType (typeof (GCPhysicalInputProfile))]
+	interface GCKeyboardInput
+	{
+		[NullAllowed, Export ("keyChangedHandler", ArgumentSemantic.Copy)]
+		GCKeyboardValueChangedHandler KeyChangedHandler { get; set; }
+
+		[Export ("anyKeyPressed")]
+		bool AnyKeyPressed { [Bind ("isAnyKeyPressed")] get; }
+
+		[Export ("buttonForKeyCode:")]
+		[return: NullAllowed]
+		GCControllerButtonInput Button (nint code);
+	}
+
+	[Mac (11, 0), iOS (14, 0), TV (14, 0)]
+	[BaseType (typeof (NSObject))]
+	interface GCMouse : GCDevice
+	{
+		[NullAllowed, Export ("mouseInput", ArgumentSemantic.Strong)]
+		GCMouseInput MouseInput { get; }
+
+		[Static]
+		[NullAllowed, Export ("current", ArgumentSemantic.Strong)]
+		GCMouse Current { get; }
+
+		[Static]
+		[Export ("mice")]
+		GCMouse[] Mice { get; }
+
+		[Notification, Field ("GCMouseDidConnectNotification")]
+		NSString DidConnectNotification { get; }
+
+		[Notification, Field ("GCMouseDidDisconnectNotification")]
+		NSString DidDisconnectNotification { get; }
+
+		[Notification, Field ("GCMouseDidBecomeCurrentNotification")]
+		NSString DidBecomeCurrentNotification { get; }
+
+		[Notification, Field ("GCMouseDidStopBeingCurrentNotification")]
+		NSString DidStopBeingCurrentNotification { get; }
+	}
+
+	delegate void GCMouseMoved (GCMouseInput mouse, float deltaX, float deltaY);
+
+	[Mac (11, 0), iOS (14, 0), TV(14, 0)]
+	[BaseType (typeof (GCControllerDirectionPad))]
+	interface GCDeviceCursor {}
+
+	[Mac (11,0), iOS (14,0), TV (14,0)]
+	[BaseType (typeof (GCPhysicalInputProfile))]
+	interface GCMouseInput
+	{
+		[NullAllowed, Export ("mouseMovedHandler", ArgumentSemantic.Copy)]
+		GCMouseMoved MouseMovedHandler { get; set; }
+
+		[Export ("scroll")]
+		GCDeviceCursor Scroll { get; }
+
+		[Export ("leftButton")]
+		GCControllerButtonInput LeftButton { get; }
+
+		[NullAllowed, Export ("rightButton")]
+		GCControllerButtonInput RightButton { get; }
+
+		[NullAllowed, Export ("middleButton")]
+		GCControllerButtonInput MiddleButton { get; }
+
+		[NullAllowed, Export ("auxiliaryButtons")]
+		GCControllerButtonInput[] AuxiliaryButtons { get; }
+	}
+
+	interface IGCDevice {}
+
+	[TV (14,0), Mac (11,0), iOS (14,0)]
+	[BaseType (typeof (NSObject))]
+	[Protocol]
+	[Model]
```

Comment:
```suggestion
	[Protocol]
```

Reply:
Aha! So we can have just `[Protocol]` if we remove `[Model]` and `[BaseType...]`! So `[Model]` and `[BaseType...]` always go together but `[Protocol]` can be independent. Thank you @dalexsoto :)